src/art/local/backend.py

```diff
@@ -675,6 +678,182 @@ def _get_wandb_run(self, model: Model) -> Run | None:
+        if step is None or step == "latest":
+            # Check both local storage and S3 (if provided) for the latest checkpoint
+            local_latest_step: int | None = None
+            s3_latest_step: int | None = None
+
+            # Get latest from local storage
+            try:
+                local_latest_step = get_model_step(model, self._path)
+                if local_latest_step == 0:
+                    # get_model_step returns 0 if no checkpoints exist
+                    local_latest_step = None
+            except Exception:
+                local_latest_step = None
+
+            # Get latest from S3 if bucket provided
+            if s3_bucket is not None:
+                from art.utils.s3_checkpoint_utils import (
+                    get_latest_checkpoint_step_from_s3,
+                )
+
+                s3_latest_step = await get_latest_checkpoint_step_from_s3(
+                    model_name=model.name,
+                    project=model.project,
+                    s3_bucket=s3_bucket,
+                    prefix=prefix,
+                )
+
+            # Determine which source has the latest checkpoint
+            if local_latest_step is None and s3_latest_step is None:
+                raise ValueError(
+                    f"No checkpoints found for {model.project}/{model.name} in local storage or S3"
+                )
+            elif local_latest_step is None:
+                resolved_step = s3_latest_step  # type: ignore[assignment]
+                if verbose:
+                    print(f"Using latest checkpoint from S3: step {resolved_step}")
+            elif s3_latest_step is None:
+                resolved_step = local_latest_step
+                if verbose:
+                    print(
+                        f"Using latest checkpoint from local storage: step {resolved_step}"
+                    )
+            elif local_latest_step >= s3_latest_step:
+                # Prefer local if equal or greater
+                resolved_step = local_latest_step
+                if verbose:
+                    print(
+                        f"Using latest checkpoint from local storage: step {resolved_step} "
+                    )
+            else:
+                resolved_step = s3_latest_step
+                if verbose:
+                    print(f"Using latest checkpoint from S3: step {resolved_step} ")
+        else:
+            resolved_step = step
+
+        # Check if checkpoint exists in the original training location
+        original_checkpoint_dir = get_step_checkpoint_dir(
+            get_model_dir(model=model, art_path=self._path), resolved_step
+        )
+
+        # Check if checkpoint exists in original location
+        if os.path.exists(original_checkpoint_dir):
+            if local_path is not None:
+                # Copy from original location to custom location
+                target_checkpoint_dir = os.path.join(local_path, f"{resolved_step:04d}")
```

Comment:
I would prefer just copying the checkpoint to whatever local_path is, so that the developer can just add that path to the artifact directly.